src/lib.rs

```diff
@@ -147,6 +149,13 @@ impl Piranha {
       .set_language(*piranha_args.language().language())
       .expect("Could not set the language for the parser.");
 
+    let temp_dir = self.write_code_snippet_to_temp();
+
+    let path_to_codebase = temp_dir
+      .as_ref()
+      .map(|t| t.path().to_str().unwrap_or_default().to_string())
+      .unwrap_or(self.piranha_arguments.path_to_codebase().to_string());
```

Comment:
So, this is kinda clever, but I think it might be more readable to have explicit if conditions checking if code snippet is present, else if path to codebase is present, else an error. The way this is written is relatively easy to miss the `unwrap_or` checking for `path_to_codebase()` as a normal alternative logic path based on configuration rather than some sort of default when `code_snippet()` is unexpectedly empty... (in fact using `path_to_codebase()` is the **common** case, no?)

As a side note: if both are passed, this logic silently ignores path to codebase and checks the code snippet only, no? Is that what we want? Maybe it's better to make sure _exactly one_ of these two arguments is set? (The alternative would be that, if both are set, then the contents of `path_to_codebase` are copied to a temp directory, then the code snippet is added [could even support multiple code snippets], then Piranha is run, but that feels like overkill right now)

Reply:
Ok. I have addressed this. 
First, I move the piranha arguments check to `PiranhaArgumentBuilder::build`. I just validate it there. I think that's a better position for it. I have also made the code_snippet related logic more explicit as u mentioned. 

I know i have had to add the idiom `path_to_codebase("/some/test/path")` at each instance where I was creating some dummy piranha arguments in the test.  This is because now invalid piranha arguments cannot be constructed :O 

Look at commits https://github.com/uber/piranha/pull/372/commits/93909191f14903f370a1eb5a58da8ab5561b18bc, https://github.com/uber/piranha/pull/372/commits/f50fdb612ee25414035a873aa53fd1aeac0666a0 and https://github.com/uber/piranha/pull/372/commits/540e83dee67a6d6f653817d8ae3240ceca6a61c4